### PR TITLE
Add option setters/getters and link to OpenGL framework

### PR DESCRIPTION
**Pull request by fkaa**
Required for https://github.com/tomaka/glutin/pull/453

### CODE REVIEW

File: src/cgl.rs [hunk 1]
```diff
@@ -103,6 +137,7 @@ pub const CORE_INTEGER_ATTRIBUTES: &'static [CGLPixelFormatAttribute] =
       kCGLPFAOpenGLProfile,
       kCGLPFAVirtualScreenCount];
 
+#[link(name = "OpenGL", kind = "framework")]
```
Comment (glennw) [inline src/cgl.rs]: Why is this needed here when it wasn't previously?

**Comment by glennw** [comment]
@fkaa Thanks! I'm no mac expert but these look fine to me, apart from the one question above.

File: src/cgl.rs (hunk 1 above)
Comment (fkaa) [inline src/cgl.rs]: I got missing symbols trying it with glutin for some reason, linking to the OpenGL framework where the CGL stuff is solved it. Might be because servo (which i'm assuming this is used for) already links it via `-framework OpenGL` in some other crate?

Comment (glennw) [inline src/cgl.rs]: @pcwalton Deferring to you since I don't know much about mac. Should the opengl framework link line be added here to the cgl bindings?

Comment (pcwalton) [inline src/cgl.rs]: Yeah, this looks right to me. Linking to the framework multiple times shouldn't hurt anything.

**Comment by glennw** [comment]
r+
